packages/web-client/app/components/card-pay/deposit-workflow/index.ts

```diff
@@ -151,21 +152,26 @@ class DepositWorkflow extends Workflow {
     }),
   ]);
   cancelationMessages = new PostableCollection([
-    new NetworkAwareWorkflowMessage({
+    // currently we only cancel because of disconnections from either wallet
+    new WorkflowMessage({
       author: cardbot,
       message:
         'It looks like your wallet(s) got disconnected. If you still want to deposit funds, please start again by connecting your wallet(s).',
       includeIf() {
-        let message = this as NetworkAwareWorkflowMessage;
-        return !message.hasLayer1Account || !message.hasLayer2Account;
+        return (
+          (this as WorkflowPostable).workflow?.cancelationReason ===
```

Comment:
There might be a way to make `this` typed more elegantly. https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function

Reply:
The ts handbook is pretty helpful, I'm finding. Thanks for pointing this out.

Reply:
It's possible to type it within the options with the expected `this`, added this in https://github.com/cardstack/cardstack/pull/1759/commits/730f1995537e753f4846706a8eb1f185f9e2c192